### DISCUSSION TITLE
有关功能的问题（菜单编辑器，多工具栏，热点，聚焦）- Question about features (menu editor, multi toolbar, hotspot, focus)

### DISCUSSION THREAD

**Discussion by GHNewbiee** [linuxdeepin/developer-center#5254] [Q&A | 问答板块]
请让我知道以下功能是否可用：

1. 内置菜单编辑器
2. 多面板/工具栏全额支持
3. 热点激活（无设置）不超过两次单击
4. 保留当前应用程序的焦点选项，即没有新应用程序从已经打开的应用程序中窃取焦点，我与之合作

提前致谢。

Please let me know if the following features are available:

1. Built-in menu editor
2. Multi panels/toolbars full support
3. Hotspot activation (no setup) in no more than two clicks
4. Keep focus option for current app ie no new apps steal the focus from the one already open and I work with

Thanks in advance.

**Comment by BLumia** [comment]
Hi, since I'm not sure what are you asking for (since the question itself is a little bit weird), let me attempt to answer those questions...

> Built-in menu editor

If "built-in" refer to the desktop environment (DDE) then there arn't such thing currently, but other menu-editor application should works since they are both following the freedesktop desktop spec.

> Multi panels/toolbars full support

DDE doen't come with multi-panel, but 3rd-party "panel" can be used as-well. `latte-dock` should work just fine.

> Hotspot activation (no setup) in no more than two clicks

All DEs I uses / used to use needs at least 3 clicks to connect a no-password Wi-Fi (hotspot), usually:

1. Click on the Wi-Fi icon on the dock/taskbar panel to make the Wi-Fi panel popup
2. Click the hotspot you tend to connect
3. Click connect

I am curious why "two clicks" is specified here...

> Keep focus option for current app

Right click the titlebar of an application, and pin it on top.

**Comment by GHNewbiee** [comment]
1. ok
2. ok
3. If I remember correct, KDE Pasma takes only 2 clicks to activate hotspot.
  In general:
  a. One click is used like on-off when previously connection is set up and a direct connection icon is present to the toolbar.
  b. Two clicks is used like: one to open hotspot panel, two to choose and connect.
4. What you describe is used when one or more windows are used for quick reference eg copy/paste values etc. The scope of that feature is misunderstood! Scope of focusing is different.
   For example, when I log in for first time in the morning I run at least 7 apps and I open another 3-5 files. Not to waste time, I start working with the first one is open. There is not time to right click etc because the other apps start opening, too and interrupt my effort to pin; start again right click etc, another app takes focus, etc.

**Reply by BLumia** [reply]
> Two clicks is used like: one to open hotspot panel, two to choose and connect.

~~You still need to click to select the hotspot from the list of available hotspots as well, so it's still 3 clicks. The connection button is only available when the hotspot is selected.~~ (edit: turns out I remembered incorrectly before. both KDE and DDE can connect hotspot within 2 clicks. See the comments below)

I'm not sure what "a direct connection icon" is tho, might check that out later. Maybe that's the one that allows you connect the hotspot with a single click (if the panel is opened)?

btw, if saving time from a single click is important, then writing a shell script to connect hotspot could be more effective. You can trigger the script with keyboard shortcut so you don't need to touch your mouse even once.

By the way. Since you also assume the hotspot is pre-configured, then both KDE and DDE will auto connect your previously connected hotspot once you logged in the desktop, thus also 0 clicks are needed.

> There is not time to right click etc...

I didn't see anything to do with window focus changed. If the only thing you need is avoid switching between windows, then a tiling window manager like i3wm or swaywm could more fit your need and it allows you to use full keyboard to do window navigation, without even touching your mouse.

**Reply by GHNewbiee** [reply]
> I didn't see anything to do with window focus changed. If the only thing you need is avoid switching between windows, then a tiling window manager like i3wm or swaywm could more fit your need and it allows you to use full keyboard to do window navigation, without even touching your mouse.

Sorry, my mistake. Windows is the not correct term. I mean apps.

**Reply by GHNewbiee** [reply]
> I'm not sure what "a direct connection icon" is tho, might check that out later. Maybe that's the one that allows you connect the hotspot with a single click (if the panel is opened)?

It is a button, like that placed on screen in Android, which I have placed on a toolbar or it is automatically placed when I setup hotspot for first time and chose a **default** connection. It is click_on-click_off like those in Android.

> btw, if saving time from a single click is important, then writing a shell script to connect hotspot could be more effective. You can trigger the script with keyboard shortcut so you don't need to touch your mouse even once.

But I need to touch the keyboard! In 2023 and at middle age I do not want to twist my fingers! In addition, memory is not any more big and strong to remember combinations. I prefer easy and simple clicks! KIS fun!

**Reply by GHNewbiee** [reply]
> By the way. Since you also assume the hotspot is pre-configured, then both KDE and DDE will auto connect your previously connected hotspot once you logged in the desktop, thus also 0 clicks are needed.

I do not have all the time wired/cable connection. So, I do not need an automatic activation of hotspot on my laptop. I need a hotspot to my mobile to give access to the laptop. I am not sure but I think that when hotspot is activated on my laptop, wireless connection is not available at the same time. So, on-off function is required.

**Reply by BLumia** [reply]
Oh sorry. I must remember incorrectly before. Actually, both KDE and DDE can use 2 clicks to connect to the hotspot. On KDE the connect button is right next to the hotspot item so you can just click that. On DDE clicking the hotspot item will directly connect the hotspot (if pre-configured / password is correct). So I guess 2 clicks are no problem :)

**Comment by GHNewbiee** [comment]
4. *...* When you pin an app, actually (by 99%) you go (ie change focus) to another app to work with, don't you? Just ask yourself why you don't pin the app, each time, you work with!?!?!?

**Reply by Decodetalkers** [reply]
what you want is more like zwl-layershell when it is setted as on_demand

 [on_demand](https://wayland.app/protocols/wlr-layer-shell-unstable-v1#zwlr_layer_surface_v1:enum:keyboard_interactivity:entry:on_demand)

it is a wayland protocol

**Reply by GHNewbiee** [reply]
Sorry, I am just a simple user. I do not have the knowledge and skills to follow the technical nature of the feature.

I just use a common sense. It is like I read a book on my desk and every friend, who wants to give me a book, puts that book above mine. So, every time I have to get the new book and place it below my book. The correct should be my friend to place the new book "directly below" my book without interrupting my reading.
